src/Nethermind/Nethermind.Runner/scripts/app.ts

```diff
@@ -0,0 +1,400 @@
+  // Update uptime text
+  updateText(upTime, formatDuration(data.uptime));
+});
+sse.addEventListener("txNodes", (e) => {
+  const data = JSON.parse(e.data) as INode[];
+  txPoolNodes = data;
+});
+sse.addEventListener("txLinks", (e) => {
+  const data = JSON.parse(e.data) as TxPool;
+  updateTxPool(data);
+});
+let lastGasLimit = 30_000_000;
+sse.addEventListener("processed", (e) => {
+  if (document.hidden) return;
+  const data = JSON.parse(e.data) as Processed;
+
+  updateText(minGas, data.minGas.toFixed(2));
+  updateText(medianGas, data.medianGas.toFixed(2));
+  updateText(aveGas, data.aveGas.toFixed(2));
+  updateText(maxGas, data.maxGas.toFixed(2));
+  updateText(gasLimit, format(data.gasLimit));
+  updateText(gasLimitDelta, data.gasLimit > lastGasLimit ? '👆' : data.gasLimit < lastGasLimit ? '👇' : '👈');
+
+  lastGasLimit = data.gasLimit;
+});
+sse.addEventListener("forkChoice", (e) => {
+
+  if (document.hidden) return;
+
+  const data = JSON.parse(e.data) as ForkChoice;
+  const number = parseInt(data.head.number, 16);
+  if (number !== 0) {
+    document.getElementById("lastestBlock").classList.remove("not-active");
+    //document.getElementById("recentBlocks").classList.remove("not-active");
+  }
+  const safe = parseInt(data.safe, 16);
+  const finalized = parseInt(data.finalized, 16);
+  updateText(headBlock, number.toFixed(0));
+  updateText(safeBlock, safe.toFixed(0));
+  updateText(finalizedBlock, finalized.toFixed(0));
+
+  updateText(safeBlockDelta, `(${(safe - number).toFixed(0)})`);
+  updateText(finalizedBlockDelta, `(${(finalized - number).toFixed(0)})`);
+
+  const block = data.head;
+
+  if (block.tx.length === 0) return;
+  // Merge tx & receipts into a single array
+  // so each element has { key, size, colorValue, ... } etc.
+  // (One data item per transaction)
+  const mergedData: TransactionReceipt[] = block.tx.map((tx, i) => {
+    const receipt = block.receipts[i];
+    return { order: i, ...tx, ...receipt };
+  });
+
+  updateText(blockExtraData, parseExtraData(block.extraData));
+  updateText(blockGasUsed, format(parseInt(block.gasUsed, 16)));
+  updateText(blockGasLimit, format(parseInt(block.gasLimit, 16)));
+  updateText(blockBlockSize, formatBytes(parseInt(block.size, 16)));
+  updateText(blockTimestamp, formatUnixTimestamp(parseInt(block.timestamp, 16)));
+  updateText(blockTransactions, format(block.tx.length));
+  updateText(blockBlobs, format(block.tx.reduce((p, c) => p + c.blobs, 0)));
+
+  // Update the D3 treemap with the merged data
+  updateTreemap<TransactionReceipt>(
+    document.getElementById("block"), // or d3.select("#treemap")
+    200,
+    parseInt(data.head.gasLimit, 16),
+      mergedData,
+      // keyFn
+      d => d.hash,
+      // orderFn
+      d => d.order,
+      // sizeFn
+      d => parseInt(d.gasUsed, 16),
+      // colorFn
+      d => parseInt(d.effectiveGasPrice, 16) * parseInt(d.gasUsed, 16)
+  );
```

Comment:
Maybe refactor `updateTreeMap` to accept an object for named parameters instead of relying on comments?

```suggestion
  updateTreemap<TransactionReceipt>(
    {
      element: document.getElementById("block"), // or d3.select("#treemap"),
      height: 200,
      totalSize: parseInt(data.head.gasLimit, 16),
      data: mergedData,
      keyFn: d => d.hash,
      orderFn: d => d.order,
      sizeFn: d => parseInt(d.gasUsed, 16),
      colorFn: d => parseInt(d.effectiveGasPrice, 16) * parseInt(d.gasUsed, 16)
    });
```